### ISSUE TITLE
SELECT max counts piles rather than cards

### ISSUE THREAD

**Issue by robartsd** [ArnoldSmith86/virtualtabletop#496]
Given a holder with a pile of cards,
```
    {
      "func": "SELECT",
      "property": "parent",
      "relation": "==",
      "value": "_holderID_",
      "max": 1
    }
```
sets the `DEFAULT` collection to a collection to a collection of all the cards in the pile.

**Comment by ArnoldSmith86** [comment]
Hmm. That's really hard to change without breaking games that rely on it.

Without having thought about it too hard, I would vote for changing this.

**Comment by ArnoldSmith86** [comment]
Hmm again. The order is currently like this:

1. Get all widgets from the `source` (or all).
2. Only keep the ones that match `property`, `relation`, `value` and `type` (or `type: pile` when `type: card` was given).
3. Only keep the first `max` widgets.
4. If `mode: add`, put the old collection behind them.
5. Replace any `pile` by its children.
6. Apply `sortBy`.

I feel like a lot of that order is not as you would expect. I would expect this:

1. Get all widgets from the `source` (or all).
2. Only keep the ones that match `property`, `relation`, `value` and `type` (or `type: pile` when `type: card` was given).
3. Replace any `pile` by its children.
4. Apply `sortBy`.
5. Only keep the first `max` widgets.
6. If `mode: add`, put the old collection behind them.

That's definitely more change than I feel comfortable with though.

We could always introduce a `legacy` property to `SELECT` and have the fileupdater add it to every affected `SELECT` and then have both behaviors in the code.

_Or_ we could split affected `SELECT`s into multiple `SELECT`s that emulate the old behavior. I'm not 100% sure that's possible.

**Comment by bjalder26** [comment]
I think this probably should have been closed with #782
